diff --git a/bulk-tcf/include/bulk_tcf_metadata.cuh b/bulk-tcf/include/bulk_tcf_metadata.cuh
index af01359..70f19bf 100644
--- a/bulk-tcf/include/bulk_tcf_metadata.cuh
+++ b/bulk-tcf/include/bulk_tcf_metadata.cuh
@@ -24,7 +24,7 @@
 #define TCF_WARPS_PER_BLOCK 16
 #define BLOCK_SIZE (TCF_WARPS_PER_BLOCK * 32)
 
-#define BLOCKS_PER_THREAD_BLOCK 64
+#define BLOCKS_PER_THREAD_BLOCK 128
 
 //# of blocks to be inserted per warp in the bulked insert phase
 //#define REGIONS_PER_WARP 8
@@ -45,7 +45,7 @@
 #define VAL_BITS 16 
 
 #define BYTES_PER_CACHE_LINE 128
-#define CACHE_LINES_PER_BLOCK 4
+#define CACHE_LINES_PER_BLOCK 2
 
 
 #endif
