diff --git a/gqf/src/gqf.cu b/gqf/src/gqf.cu
index d37e6d7..6cb84cb 100644
--- a/gqf/src/gqf.cu
+++ b/gqf/src/gqf.cu
@@ -157,7 +157,7 @@ __host__ __device__ static inline int popcnt(uint64_t val)
 #else
 
 	#ifndef __x86_64
-		val = __builtin_popcount(val);
+		val = __builtin_popcountll(val);
 
 	#else
 		
@@ -206,7 +206,7 @@ __host__ __device__ static inline int bitrank(uint64_t val, int pos) {
 
 	#ifndef __x86_64
 
-		val = __builtin_popcount(val);
+		val = __builtin_popcountll(val);
 
 	#else
 
